packages/components/checkbox/src/checkbox.js

```diff
@@ -36,18 +36,17 @@ export class TSCheckbox extends TSElement {
 
 	render() {
 		return html`
-			<label class="checkbox-container">
-				${this.label}
+			<div class="checkbox-container" @click="${this.onClick}">
 				<input
 					type="checkbox"
-					@click="${this.onClick}"
 					.name="${this.name}"
 					.value="${this.value}"
 					?checked="${this.checked}"
 					?disabled="${this.disabled}"
 				/>
-				<span></span>
-			</label>
+				<div class="checkbox"></div>
+				${this.label}
```

Comment:
Since we are not forcing to have text for the `label`, I think it does make sense to have slot for it to be customizable

Reply:
I'm not sure what do you mean by this, can you explain a bit more? Currently we don't have any slots and do not support any customisation for a label.

Reply:
Since developers can just don't provide text for the label and put it next some other stuff like an image or icon, then maybe we can replace attribute with slot in this case, to let them put icon, image, etc. in the slot, so they can have the benefit of triggering checkbox by click on the label as well.

Reply:
Right now, they can just put an empty checkbox and next to it, use an image, and with some custom js get the similar functionality

Reply:
Ok, I see now. But how checkbox should be vertically aligned with this content? Also, I think we need to discuss it with UI/UX team first, do we want to provide such functionality.
And I suggest to implement it in a separate PR, since it will be a feature, not a fix.

Reply:
Sure we can do it in a separate PR 👍 
But it's not a feature to provide, it's already achievable, only in a bit hacky way.

Reply:
I mean, that In terms of standard `ts-checkbox` functionality it will be considered as feature, because right now you cannot achieve this behaviour declaratively, without additional code on application side.
Of course you can implement many additional features on application side, but it doesn't mean that it's `native` implementation should be considered as a fix :)